ext/storage_sources/azure_store/azure_storage_source.cpp

```diff
@@ -87,26 +90,88 @@ static int azure_file_read(WT_FILE_HANDLE *, WT_SESSION *, wt_off_t, size_t, voi
   __attribute__((__unused__));
 static int azure_file_size(WT_FILE_HANDLE *, WT_SESSION *, wt_off_t *) __attribute__((__unused__));
 
+// Return a customised file system to access the Azure storage source.
 static int
 azure_customize_file_system(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session,
-  const char *bucket_name, const char *auth_token, const char *config,
-  WT_FILE_SYSTEM **file_systemp)
+  const char *bucket, const char *auth_token, const char *config, WT_FILE_SYSTEM **file_system)
 {
-    WT_UNUSED(storage_source);
-    WT_UNUSED(session);
-    WT_UNUSED(bucket_name);
-    WT_UNUSED(auth_token);
-    WT_UNUSED(config);
-    WT_UNUSED(file_systemp);
-
-    return 0;
+    if (bucket == nullptr || std::string(bucket).length() == 0) {
+        std::cerr << "azure_customize_file_system: Bucket not specified." << std::endl;
+        return EINVAL;
+    }
+    // Parse config string.
+
+    // Get any prefix to be used for the object keys.
+    WT_CONFIG_ITEM obj_prefix_config;
+    std::string obj_prefix;
+
+    // Get the value of the config key from the string
+    azure_store *azure = (azure_store *)storage_source;
+    int ret;
+    if ((ret = azure->wt_api->config_get_string(
+           azure->wt_api, session, config, "prefix", &obj_prefix_config)) == 0)
+        obj_prefix = std::string(obj_prefix_config.str, obj_prefix_config.len);
+    else if (ret != WT_NOTFOUND) {
+        std::cerr << "azure_customize_file_system: error parsing config for object prefix."
+                  << std::endl;
+        return ret;
+    }
+
+    // Fetch the native WT file system.
+    WT_FILE_SYSTEM *wt_file_system;
+    if ((ret = azure->wt_api->file_system_get(azure->wt_api, session, &wt_file_system)) != 0)
+        return ret;
+
+    // Get a copy of the home directory.
+    const std::string home_dir = session->connection->get_home(session->connection);
+
+    // Create the file system.
+    azure_file_system *fs;
+    if ((fs = (azure_file_system *)calloc(1, sizeof(azure_file_system))) == nullptr) {
+        std::cerr << "azure_customize_file_system: unable to allocate memory for file system."
+                  << std::endl;
+        return ENOMEM;
+    }
+
+    fs->store = azure;
+    fs->wt_fs = wt_file_system;
+    fs->home_dir = home_dir;
+
+    try {
+        fs->azure_conn = new azure_connection(bucket, obj_prefix);
```

Comment:
Usually we want to avoid new/delete c++ function because it is bad practice. Let's make this a unique pointer, use the concept of std::move.